### PR TITLE
Add missing info in API doc

### PR DESCRIPTION
**Pull request by galpeter**
As a few methods were removed and replaced it is a good idea to describe
which method(s) can reproduce the same functionality. In addition add a
bit of extra comments for various methods/structs.

JerryScript-DCO-1.0-Signed-off-by: Peter Gal pgal.usz@partner.samsung.com

### CODE REVIEW

File: docs/02.API-REFERENCE.md [hunk 1]
```diff
@@ -686,7 +692,7 @@ typedef struct jerry_call_info_t
 } jerry_call_info_t;
 ```
 
-*New in version [[NEXT_RELEASE]]*.
+*New in version [[NEXT_RELEASE]]*. Contest of this struct replaces the `jerry_get_new_target` function.
```
Comment (rerobika) [inline docs/02.API-REFERENCE.md:695]: I think you meant content instead of contest.

File: docs/02.API-REFERENCE.md [hunk 2]
```diff
@@ -243,7 +243,7 @@ Option bits for [jerry_parse_options_t](#jerry_parse_options_t).
  - JERRY_PARSE_NO_OPTS - No options passed
  - JERRY_PARSE_STRICT_MODE - Enable strict mode
  - JERRY_PARSE_MODULE - Parse source as an ECMAScript module
- - JERRY_PARSE_HAS_ARGUMENT_LIST - `argument_list` field is valid
+ - JERRY_PARSE_HAS_ARGUMENT_LIST - `argument_list` field is valid, this also means that function parsing will be done
```
Comment (zherczeg) [inline docs/02.API-REFERENCE.md:246]: Update the comment in the include/types.h as well (these kept in sync).

**Review by zherczeg — APPROVED** [review]
LGTM with minor change

File: docs/02.API-REFERENCE.md (hunk 2 above)
Comment (galpeter) [reply docs/02.API-REFERENCE.md:246]: Good idea. I'll update the related header file.

**Comment by galpeter** [comment]
Update done and also added an extra comment for the parse option bits.

**Review by rerobika — APPROVED** [review]

File: docs/02.API-REFERENCE.md [hunk 3]
```diff
@@ -67,7 +67,7 @@ Enum that contains JerryScript **object** value types:
 
 *New in version 2.4*.
 
-*Changed in version [[NEXT_RELEASE]]*: Added `JERRY_OBJECT_TYPE_SCRIPT`, `JERRY_OBJECT_TYPE_MODULE` values.
+*Changed in version [[NEXT_RELEASE]]*: Added `JERRY_OBJECT_TYPE_WEAKREF`, `JERRY_OBJECT_TYPE_SCRIPT`, `JERRY_OBJECT_TYPE_MODULE` values.
```
Comment (rerobika) [inline docs/02.API-REFERENCE.md:70]: How about `JERRY_OBJECT_TYPE_SHARED_ARRAYBUFFER`? I think it need to be mentioned as well.

Comment (galpeter) [reply docs/02.API-REFERENCE.md:70]: True.

**Comment by rerobika** [comment]
There were a lot of api doc changes nowdays, please rebase.

**Comment by galpeter** [comment]
Rebased, as per request.

**Pull request merged**